python/tests/test_topology.py

```diff
@@ -3466,6 +3460,282 @@ def test_many_trees_recurrent_mutations_internal_samples(self):
+    2       1       0               0.00000000000000
+    3       1       0               0.00000000000000
+    4       1       0               0.00000000000000
+    5       0       0               0.14567111023387
+    6       0       0               0.21385545626353
+    7       0       0               0.43508024345063
+    8       0       0               1.60156352971203
+    """
+    edges = """\
+    id      left            right           parent  child
+    0       0.00000000      1.00000000      5       0,1
+    1       0.00000000      1.00000000      6       2,3
+    2       0.00000000      1.00000000      7       4,5
+    3       0.00000000      1.00000000      8       6,7
+    """
+    #
+    #          9                        10
+    #         / \                      / \
+    #        /   \                    /   8
+    #       /     \                  /   / \
+    #      7       \                /   /   \
+    #     / \       6              /   /     6
+    #    /   5     / \            /   5     / \
+    #   /   / \   /   \          /   / \   /   \
+    #  4   0   1 2     3        4   0   1 2     3
+    #
+    # 0 ------------------ 0.5 ------------------ 1.0
+    nodes0 = """\
+    id      is_sample   population      time
+    0       1       0               0.00000000000000
+    1       1       0               0.00000000000000
+    2       1       0               0.00000000000000
+    3       1       0               0.00000000000000
+    4       1       0               0.00000000000000
+    5       0       0               0.14567111023387
+    6       0       0               0.21385545626353
+    7       0       0               0.43508024345063
+    8       0       0               0.60156352971203
+    9       0       0               0.90000000000000
+    10      0       0               1.20000000000000
+    """
+    edges0 = """\
+    id      left            right           parent  child
+    0       0.00000000      1.00000000      5       0,1
+    1       0.00000000      1.00000000      6       2,3
+    2       0.00000000      0.50000000      7       4,5
+    3       0.50000000      1.00000000      8       5,6
+    4       0.00000000      0.50000000      9       6,7
+    5       0.50000000      1.00000000      10      4,8
+    """
+    nodes1 = """\
+    id      is_sample   population      time
+    0       0           0           1.0
+    1       1           0           0.0
+    2       1           0           0.0
+    """
+    edges1 = """\
+    id      left            right           parent  child
+    0       0.00000000      1.00000000      0       1,2
+    """
+
+    def do_map(self, ts, ancestors, samples=None, compare_lib=True):
+        """
+        Runs the Python test implementation of map_ancestors.
+        """
+        if samples is None:
+            samples = ts.samples()
+        s = tests.AncestorMap(ts, samples, ancestors)
+        ancestor_table = s.simplify()
```

Comment:
I was confused that the "map ancestors" method of `AncestorMap` is called `.simplify()`. Perhaps this should be called `.map_ancestors()` also? I'm trying to keep the tests readable.